backend/src/app/services/snapshot/snapshot.service.ts

```diff
@@ -151,6 +152,121 @@ export class SnapshotsService {
     }
   }
 
+  getNotatioParticipantsForSnapshotCreation = async (notationId: string) => {
+    try {
+      if (notationId === '' || notationId === null) {
+        throw Error('notation id cannot be empty');
+      }
+
+      return await this.eventsParticipantsRepo.find({
+        where: {
+          eventId: notationId,
+          srAction: SRApprovalStatusEnum.PUBLIC,
+        },
+      });
+    } catch (error) {
+      throw error;
```

Comment:
implement logging in getNotatioParticipantsForSnapshotCreation() - start, end and catch block of the method. Also refer to the HttpExecption implemented in the other service layers and check the feasibility of having it.

Reply:
Added log statements to all the method. As discussed HTTP exception is not required here as its already handled in main service method with proper response code.